scopes/component/capsules-syncer/capsules-syncer.task.ts

```diff
@@ -0,0 +1,43 @@
+import { BuildContext, BuiltTaskResult, BuildTask } from '@teambit/builder';
+import path from 'path';
+import { CompilerAspect } from '@teambit/compiler';
+import { DevFilesMain } from '@teambit/dev-files';
+import { DependencyResolverMain } from '@teambit/dependency-resolver';
+import componentIdToPackageName from '@teambit/legacy/dist/utils/bit/component-id-to-package-name';
+import { hardLinkDirectory } from '@teambit/toolbox.fs.hard-link-directory';
+
+export class CapsulesSyncerTask implements BuildTask {
```

Comment:
I don't think this should be a task for a few reasons:
1. we don't want to run it only on build (we might create capsules on other flows, like `bit capsule create` command
2. when it's a build task like this, you need to register it from the env, so every env developer should be aware of it. 
3. logically it's not a build task, it's part of the capsule creation. it's part of the core of the isolation. like writing the files or the installation is not a build task.
4. let's move all of this into the isolator (we can have it as a different component, but it shouldn't be an aspect I think. as there is no real reason to make it an aspect. (we just need to provide the hasRootComponents and injected dirs from outside

Reply:
But it should happen after the compile tasks finish and before the tester task starts. Currently there is no other way to do this. All tasks are executed in sequence. There is no code that runs between the compile and test tasks. The only way to run code between them is to add a new task.

Reply:
@zkochan in the compiler task here - https://github.com/teambit/bit/blob/d832fe8cdf455d507c2ae87a74961c6fbfaa91b5/scopes/compilation/compiler/compiler.task.ts#L30
we get the build results, which contain all the artifacts. why not adding all this logic there inside this function? get the build results, run the artifacts glob, read from the fs, then write to the injected dirs
will be much more clean. the env won't need to be aware of it. it will be atomic as part of the compilation step, so less room for mistakes by someone remove it, forget it, or change the order.
@davidfirst what do you think?

Reply:
@GiladShoham , this `execute` is running per env, so it can be running multiple times during `bit build`. not sure if that's fine.
but yeah, in general, it's better to put it in the compiler aspect than as a separate build-task if possible.

Reply:
ok, I have removed the capsules syncer task.